include/materials/H5MoltresMaterial.h

```diff
@@ -0,0 +1,119 @@
+#ifndef H5MOLTRESMATERIAL_H_
+#define H5MOLTRESMATERIAL_H_
+
+#include "GenericConstantMaterial.h"
+#include "SplineInterpolation.h"
+#include "BicubicSplineInterpolation.h"
+#include "MonotoneCubicInterpolation.h"
+#include "LinearInterpolation.h"
+#include "H5Cpp.h"
+
+class H5MoltresMaterial;
+
+template <>
+InputParameters validParams<H5MoltresMaterial>();
+
+/**
+ * This class hinges on the `interp_type` chosen by the user. The user can choose from the
+ * following interpolation types:
+ *
+ * "normal interpolations"
+ * spline :         cubic spline interpolation for single temperature. Example input property
+ *                  table: property_file_dir/newt_msre_fuel_NSF.txt
+ * monotone_cubic : monotone cubic interpolation for single temperature. Same input property table
+ *                  structure as spline
+ * linear :         linear interpolation for single temperature. Same input property table
+ *                  structure as spline
+ *
+ * "special interpolations"
+ * none :           no interpolation is done. The cross section value for each group is read from
+ *                  the first temperature row in the interpolation table
+ */
+class H5MoltresMaterial : public GenericConstantMaterial
```

Comment:
Why not inherit from MoltresMaterial instead? One would have to make some MoltresMaterial functions inheritable but then would only have to adjust the methods that are different for HFMoltresMaterial.